test/Microsoft.AspNet.Mvc.Formatters.Json.Test/JsonResultTest.cs

```diff
@@ -21,155 +18,24 @@ namespace Microsoft.AspNet.Mvc
-
-            // Act
-            await result.ExecuteResultAsync(actionContext);
-            var written = GetWrittenBytes(context);
-
-            // Assert
-            Assert.Equal(expected, written);
-            Assert.Equal("text/json", context.Response.ContentType);
-        }
-
-        [Fact]
-        public async Task ExecuteResultAsync_SetsContentTypeAndEncoding()
-        {
-            // Arrange
-            var expected = _abcdUTF8Bytes;
-
-            var context = GetHttpContext();
-            var actionContext = new ActionContext(context, new RouteData(), new ActionDescriptor());
-
-            var result = new JsonResult(new { foo = "abcd" });
-            result.ContentType = new MediaTypeHeaderValue("text/json")
-            {
-                Encoding = Encoding.ASCII
-            };
-
-            // Act
-            await result.ExecuteResultAsync(actionContext);
-            var written = GetWrittenBytes(context);
-
-            // Assert
-            Assert.Equal(expected, written);
-            Assert.Equal("text/json; charset=us-ascii", context.Response.ContentType);
-        }
-
         [Fact]
-        public async Task NoResultContentTypeSet_UsesResponseContentType_AndSuppliedEncoding()
+        public async Task ExecuteAsync_WritesJsonContent()
         {
             // Arrange
-            var expectedData = Encoding.ASCII.GetBytes("{\"foo\":\"abcd\"}");
-            var expectedContentType = "text/foo; p1=p1-value; charset=us-ascii";
-            var context = GetHttpContext();
-            context.Response.ContentType = expectedContentType;
-            var actionContext = new ActionContext(context, new RouteData(), new ActionDescriptor());
+            var value = new { foo = "abcd" };
+            var expected = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(value));
 
-            var result = new JsonResult(new { foo = "abcd" });
+            var context = GetActionContext();
 
-            // Act
-            await result.ExecuteResultAsync(actionContext);
-            var written = GetWrittenBytes(context);
-
-            // Assert
-            Assert.Equal(expectedData, written);
-            Assert.Equal(expectedContentType, context.Response.ContentType);
-        }
-
-        [Theory]
-        [InlineData("text/foo", "text/foo")]
-        [InlineData("text/foo; p1=p1-value", "text/foo; p1=p1-value")]
-        public async Task NoResultContentTypeSet_UsesResponseContentTypeAndDefaultEncoding_DoesNotSetCharset(
-            string responseContentType,
-            string expectedContentType)
-        {
-            // Arrange
-            var expected = _abcdUTF8Bytes;
-
-            var context = GetHttpContext();
-            context.Response.ContentType = responseContentType;
-            var actionContext = new ActionContext(context, new RouteData(), new ActionDescriptor());
-
-            var result = new JsonResult(new { foo = "abcd" });
+            var result = new JsonResult(value);
```

Comment:
Leaving one test behind to verify the end to end. The details are tested in the tests for the executor.